rmw_dds_common/src/security.cpp

```diff
@@ -40,20 +40,27 @@ bool get_security_files(
   };
 
   for (const std::pair<const std::string, std::string> & el : required_files) {
-    rcpputils::fs::path full_path(secure_root);
+    std::filesystem::path full_path(secure_root);
     full_path /= el.second;
-    if (!full_path.is_regular_file()) {
+    std::filesystem::directory_entry target(full_path);
+    if (!target.is_regular_file()) {
       result.clear();
       return false;
     }
 
     result[el.first] = prefix + full_path.string();
   }
 
+  // Check all required files are exists
+  if (result.size() != required_files.size()) {
+    return false;
+  }
+
   for (const std::pair<const std::string, std::string> & el : optional_files) {
-    rcpputils::fs::path full_path(secure_root);
+    std::filesystem::path full_path(secure_root);
     full_path /= el.second;
-    if (full_path.is_regular_file()) {
+    std::filesystem::directory_entry target(full_path);
+    if (target.is_regular_file()) {
```

Comment:
Same thing here, I don't think we need this extra `directory_entry`:

```suggestion
    if (std::filesystem::is_regular_file(full_path)) {
```

Reply:
Yes. I changed it in the last commit.